pceplib/pcep_msg_object_error_types.c

```diff
@@ -0,0 +1,403 @@
+/*
+ * This file is part of the PCEPlib, a PCEP protocol library.
+ *
+ * Copyright (C) 2020 Volta Networks https://voltanet.io/
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ * Author : Brady Johnson <brady@voltanet.io>
+ */
+
+#include <stdlib.h>
+
+#include "pcep_msg_object_error_types.h"
+#include "pcep_utils_logging.h"
+
+/* All of these values were copied from:
+ *     https://www.iana.org/assignments/pcep/pcep.xhtml#pcep-error-object
+ * Which was last updated 2020-06-02 */
+
+static const char *error_type_strings[] =
+{
+        "Reserved",
+        "PCEP session establishment failure",
+        "Capability not supported",
+        "Unknown Object",
+        "Not supported object",
+        "Policy violation",
+        "Mandatory Object missing",
+        "Synchronized path computation request missing",
+        "Unknown request reference",
+        "Attempt to establish a second PCEP session",
+
+        "Reception of an invalid object",  /* 10 */
+        "Unrecognized EXRS subobject",
+        "Diffserv-aware TE error",
+        "BRPC procedure completion failure",
+        "Unassigned 14",
+        "Global Concurrent Optimization Error",
+        "P2MP Capability Error",
+        "P2MP END-POINTS Error",
+        "P2MP Fragmentation Error",
+        "Invalid Operation",
+
+        "LSP State Synchronization Error",   /* 20 */
+        "Invalid traffic engineering path setup type",
+        "Unassigned 22",
+        "Bad parameter value",
+        "LSP instantiation error",
+        "PCEP StartTLS failure",
+        "Association Error",
+        "WSON RWA Error",
+        "H-PCE Error",
+        "Path computation failure",
+        "Unassigned 30"
+};
+
+/* Store how many Error Values each Error Type has */
+static const int error_value_lengths[] =
+{
+    0,  8,  0,  2,  9,  /* Types 0 - 4 */
+    8, 15,  0,  0,  0,
+    32, 0,  3,  1,  0,  /* Types 10 - 14 */
+    2,  2,  4,  4, 14,
+    7,  2,  0,  2,  3,  /* Types 20 - 24 */
+    5, 11,  3,  2,  4
+};
+
+static const char *error_value_strings[MAX_ERROR_TYPE][MAX_ERROR_VALUE] =
+{
+
+    /* 0   Reserved */
+    { "Reserved"},
```

Comment:
Since this array is of size 10 anyway, why not put "Unassigned" as the remainder of the values and ditch the error-prone `error_value_lengths` array?

Reply:
done